File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/di/scheduler/RxImagePickerTestSchedulers.java
package com.qingmei2.rximagepicker.di.scheduler;

import android.support.annotation.RestrictTo;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

import static android.support.annotation.RestrictTo.Scope.TESTS;

/**
 * An extra layer of packaging, using {@link RxImagePickerTestSchedulers} for easier testing.
 */
@RestrictTo(TESTS)
public class RxImagePickerTestSchedulers implements IRxImagePickerSchedulers {

    @Override
    public Scheduler ui() {
        return Schedulers.io();
    }

    @Override
    public Scheduler io() {
        return Schedulers.io();
    }
}

File: rximagepicker/src/main/java/com/qingmei2/rximagepicker/scheduler/RxImagePickerTestSchedulers.kt
package com.qingmei2.rximagepicker.scheduler

import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.schedulers.Schedulers

/**
 * An extra layer of packaging, using [RxImagePickerTestSchedulers] for easier testing.
 */
class RxImagePickerTestSchedulers : IRxImagePickerSchedulers {

    override fun ui(): Scheduler {
        return Schedulers.io()
    }

    override fun io(): Scheduler {
        return Schedulers.io()
    }
}
